File: mirea/cources/2026-2027/aii-august/08/contest/A.py
from math import gcd, lcm
def solve():
    # читаем данные:
    g, l = map(int, input().split())
    # перебираем все подходящие значения a, проверяя каждое:
    answ = []
    for a in range(1, 1_000_001):
        if a % g == 0 and l % a == 0:
            b = g * l // a # используем a * b = НОК * НОД
            if b <= 1_000_000 and gcd(a, b) == g and lcm(a, b) == l:
                answ.append((a, b))
    # выводим ответ:
    print(len(answ))
    for a, b in answ:
        print(a, b)
tt = int(input())
for _ in range(tt):
    solve()

File: mirea/cources/2026-2027/aii-august/08/contest/A.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
void solve() {
    // читаем данные:
    ll g, l; cin >> g >> l;
    // перебираем все подходящие значения a, проверяя каждое:
    vector<pair<ll,ll>> answ;
    for (ll a = 1; a <= 1'000'000; a++)
        if (a % g == 0 && l % a == 0) {
            ll b = g * l / a; // используем a * b = НОК * НОД
            if (b <= 1'000'000 && gcd(a,b) == g && lcm(a,b) == l)
                answ.emplace_back(a, b);
        }
    // выводим ответ:
    cout << answ.size() << '\n';
    for (auto &[a, b] : answ)
        cout << a << ' ' << b << '\n';
}
main() {
    int tt; cin >> tt;
    while (tt --> 0) solve();
}
